Filrb/Internal/Model.lean

```diff
@@ -169,27 +197,181 @@ theorem bst_iff_sorted_inorder {t : Raw α} : t.BST ↔ Sorted t.inorder := by
 namespace Raw
 namespace Model
 
-omit [Preorder α] [Ord α] [LawfulOrd α] in
+omit [Ord α] [LawfulOrd α] in
 @[simp]
-theorem inorder_paintColor_independent (t : Raw α) :
-    (t.paintColor c).inorder = t.inorder := by
+lemma inorder_paintColor_independent (t : Raw α ) : (t.paintColor c).inorder = t.inorder := by
   unfold paintColor
   split <;> simp
 
-theorem baliL_inorder_independent {l r : Raw α}
-    (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l)
-    (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r) :
-    (baliL x l r).inorder = l.inorder ++ x :: r.inorder := by
-  sorry
+omit [Ord α] [LawfulOrd α] in
+lemma baliL_inorder_independent{l r : Raw α} (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l) (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r)
+    : (baliL x l r).inorder = l.inorder ++ x :: r.inorder := by
+    simp[baliL]
+    split <;> aesop
 
-theorem baliR_inorder_independent {l r : Raw α}
-    (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l)
-    (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r) :
-    (baliR x l r).inorder = l.inorder ++ x :: r.inorder := by
-  sorry
+omit [Ord α] [LawfulOrd α] in
+lemma baliR_inorder_independent {l r : Raw α} (hl1 : ∀ y ∈ l, y < x) (hl2 : BST l) (hr1 : ∀ y ∈ r, x < y) (hr2 : BST r)
+    : (baliR x l r).inorder = l.inorder ++ x :: r.inorder := by
+    simp[baliR]
+    split <;> aesop
+
+lemma sortedInsert_left (x data : α) (xs ys : List α) (hl1 : ∀ a ∈ xs, a < data ) (hr1 : ∀ b ∈ ys, data < b) (h : x < data) :
+  sortedInsert xs x ++ data :: ys = sortedInsert (xs ++ data :: ys) x := by
```

Comment:
The theorem statement on the separate line should be indented by 4 not 2 spaces, additionally the first line is also longer than 120 characters.